docs/source/services-basics.md

```diff
@@ -1,36 +1,131 @@
-## External services
+# External services
 
-JupyterHub has a REST API that can be used by external services like the
-[cull_idle_servers](https://github.com/jupyterhub/jupyterhub/blob/master/examples/cull-idle/cull_idle_servers.py)
-script which monitors and kills idle single-user servers periodically. In order to run such an
-external service, you need to provide it an API token. In the case of `cull_idle_servers`, it is passed
-as the environment variable called `JPY_API_TOKEN`.
+When working with JupyterHub, a **Service** is defined as a process
+that interacts with the Hub's REST API. A Service may perform a specific
+or action or task. For example, shutting down individuals' single user
+notebook servers that have been is a good example of a task that could
+be automated by a Service. Let's look at how the [cull_idle_servers][]
+script can be used as a Service.
 
-Currently there are two ways of registering that token with JupyterHub. The first one is to use
-the `jupyterhub` command to generate a token for a specific hub user:
+## Real-world example to cull idle servers
+
+JupyterHub has a REST API that can be used by external services. This
+document will:
+
+- explain some basic information about API tokens
+- clarify that API tokens can be used to authenticate to
+  single-user servers as of [version 0.8.0](./changelog.html)
+- show how the [cull_idle_servers][] script can be:
+    - used in a Hub-managed service
+    - run as a standalone script
+
+Both examples for `cull_idle_servers` will communicate tasks to the
+Hub via the REST API.
+
+## API Token basics
+
+### Create an API token
+
+To run such an external service, an API token must be created and
+provided to the service.
+
+As of [version 0.6.0](./changelog.html), the preferred way of
+generating an API token is:
 
 ```bash
-jupyterhub token <username>
+openssl rand -hex 32
 ```
 
-As of [version 0.6.0](./changelog.html), the preferred way of doing this is to first generate an API token:
+Alternatively, use the `jupyterhub token` (*deprecated in version 0.8.0*)
+command to generate a token for a specific hub user:
 
 ```bash
-openssl rand -hex 32
+jupyterhub token <username>
 ```
 
+### Pass environment variable with token to the Hub
+
+In the case of `cull_idle_servers`, it is passed as the environment
+variable called `JPY_API_TOKEN`.
+
+### Pass token using a configuration file
 
-and then write it to your JupyterHub configuration file (note that the **key** is the token while the **value** is the username):
+You can write the API token to your JupyterHub
+configuration file, `jupyterhub_config.py` (note that
+the **key** is the 'token' while the **value** is the 'username'):
 
 ```python
 c.JupyterHub.api_tokens = {'token' : 'username'}
```

Comment:
Let's remove this mention of using `.api_tokens` for services, and leave only the `.services` config below.